presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -92,17 +97,21 @@ public PagesSpatialIndexSupplier(
         this.filterFunctionFactory = filterFunctionFactory;
         this.partitions = partitions;
 
-        this.rtree = buildRTree(addresses, channels, geometryChannel, radiusChannel, partitionChannel);
+        this.rtree = buildRTree(addresses, channels, geometryChannel, radiusChannel, partitionChannel, localUserMemoryContext);
         this.radiusChannel = radiusChannel;
         this.memorySizeInBytes = INSTANCE_SIZE +
                 (rtree.isEmpty() ? 0 : STRTREE_INSTANCE_SIZE + computeMemorySizeInBytes(rtree.getRoot()));
     }
 
-    private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> channels, int geometryChannel, Optional<Integer> radiusChannel, Optional<Integer> partitionChannel)
+    private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> channels, int geometryChannel, Optional<Integer> radiusChannel, Optional<Integer> partitionChannel, LocalMemoryContext localUserMemoryContext)
     {
         STRtree rtree = new STRtree();
         Operator relateOperator = OperatorFactoryLocal.getInstance().getOperator(Operator.Type.Relate);
 
+        long existingMemorySizeInBytes = localUserMemoryContext.getBytes();
```

Comment:
I think 2 variables would be sufficient: one to track last reported size and one to track memory added since then

```
        long recordedSizeInBytes = localUserMemoryContext.getBytes();
        long addedSizeInBytes = STRTREE_INSTANCE_SIZE;
```